app/gui/view/welcome-screen/src/side_menu.rs

```diff
@@ -58,4 +86,37 @@ impl SideMenu {
         projects_list.append_or_warn(&button, logger);
         button
     }
+
+    fn setup_new_project_event_listener(&self) {
+        let frp = self.frp.clone_ref();
```

Comment:
this is a potential memory leak. We should disable cloning of FRP. The struct that you are cloning contains the whole FRP graph (see the docs in FRP to learn more). Instead, you should clone here only the endpoint that you will be using (which contains weak reference to the graph), namely the `frp.create_project`.

Reply:
Thank you, I wasn't aware of it. I changed it to passing endpoints everywhere.